website/content/docs/auth/jwt/oidc_providers.mdx

```diff
@@ -262,3 +262,148 @@ vault write auth/oidc/role/your_default_role \
 1. Save client ID and secret.
 
 Note your policy will need `oidc_scopes` to include `profile` to get a full profile ("[Fat Token](https://support.okta.com/help/s/article/Okta-Groups-or-Attribute-Missing-from-Id-Token)"). You will also need to configure bound audience along the lines of `"bound_audiences": ["api://default", "0a4........."]` if you are using the default authorization server.
+
+## Kubernetes
+
+Kubernetes can function as an OIDC provider such that Vault can validate its
+service account tokens using JWT/OIDC auth.
+
+-> **Note:** The JWT auth engine does **not** use Kubernetes' `TokenReview` API
+during authentication, and instead uses public key cryptography to verify the
+contents of JWTs. This means tokens that have been revoked by Kubernetes will
+still be considered valid by Vault until their expiry time. To mitigate this
+risk, use short TTLs for service account tokens or use
+[Kubernetes auth][/docs/auth/kubernetes] which _does_ use the `TokenReview` API.
+
+### Using service account issuer discovery
+
+Kubernetes cluster requirements:
+
+* `ServiceAccountIssuerDiscovery` feature gate enabled.
+   * Present from 1.18, defaults to true from 1.20.
+* Must use short-lived service account tokens when logging in.
+   * Tokens mounted into pods default to short-lived from 1.21.
+* kube-apiserver's `--service-account-issuer` flag is set to a URL that is
+  reachable from Vault. Public by default for most managed Kubernetes solutions.
+
+1. Ensure OIDC discovery URLs do not require authentication, as detailed
+   [here](k8s-sa-issuer-discovery):
+
+   ```bash
+   kubectl create clusterrolebinding oidc-reviewer  \
+      --clusterrole=system:service-account-issuer-discovery \
+      --group=system:unauthenticated
+   ```
+
+1. Enable and configure JWT auth in Vault.
+
+   1. If Vault is running in Kubernetes:
+
+      ```bash
+      kubectl exec vault-0 -- vault auth enable jwt
+      kubectl exec vault-0 -- vault write auth/jwt/config \
+         oidc_discovery_url=https://kubernetes.default.svc.cluster.local \
+         oidc_discovery_ca_pem=@/run/secrets/kubernetes.io/serviceaccount/ca.crt
+      ```
+
+   1. Alternatively, if Vault is _not_ running in Kubernetes:
+
+      -> **Note:** When Vault is outside the cluster, the `$ISSUER` below may
+         or may not be reachable. If not, you can configure JWT auth using
+         [`jwt_validation_pubkeys`](#using-jwt-validation-public-keys) instead.
+
+      ```bash
+      # 1. Find the issuer URL of the cluster.
```

Comment:
This approach speculates that every cluster can obtain the information through this endpoint. This is different from the current "[discovering the service account issuer](https://www.vaultproject.io/docs/auth/kubernetes#discovering-the-service-account-issuer)" block. Can this be changed?

Reply:
@benashz My apologies for not being clearer. With "the block" I was talking about the following text in the "discovering the service account issuer" section in the Kubernetes Auth docs:
```
Most clusters will also have that information available at the .well-known/openid-configuration endpoint:
```

Here it states that _most_ clusters will also have that information available rather than always having this available. 

I don't particularly see this difference as a downside, perhaps even a bit picky, which is also the reason why I didn't put this under "Request change". Rather this might be a nice to have as the documentation in that case is identical.

Reply:
If `ServiceAccountIssuerDiscovery` is enabled, then the cluster does have the openid-configuration endpoint available: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-issuer-discovery.

Given we have that as a requirement at the beginning of the section, I think we're covered here. If users have problems using the issuer URL themselves then I think they're likely to run into problems later when Vault tries to use it too, so I'm not too inclined to link to an alternative method either. I've cleared up the language below to make clear you don't necessarily need the issuer URL for public keys.